src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -57,34 +57,47 @@
  *     X509 CERTIFICATE, X509 CRL, and ENCRYPTED PRIVATE KEY.
  * </pre>
  *
- * A specified return class must implement {@link DEREncodable} and be an
- * appropriate JCE object class for the PEM; otherwise an
- * {@link IllegalArgumentException} is thrown.
- *
  * <p> If the PEM does not have a JCE object representation, it returns a
  * {@link PEMRecord}. Any PEM can be decoded into a {@code PEMRecord} if the
  * class is specified.
  *
- * <p> A new immutable {@code PEMDecoder} instance is created when configured
+ * For decode methods that accept a {@code Class<S> tClass} as input, they can
+ * modify the return type to a specific {@code DEREncodable} subclass.
+ * For example, {@code ECPublicKey.class} can be used to cast a
```

Comment:
The "modify" and "cast" words seem to reveal some implementation detail. It's just "parsing the input as the specified type".